### PR TITLE
Allow parameter references

### PR DESCRIPTION
**Pull request by sandermarechal**
Allow reusable parameters using refs, as specified in https://github.com/OAI/OpenAPI-Specification/blob/master/fixtures/v2.0/json/resources/reusableParameters.json

### CODE REVIEW

**Comment by sandermarechal** [comment]
@GuilhemN Can you take a look at this? The only problem I found so far is that the Parameters collection insists on using <name>/<in> as the unique identifier. That resolves to "/" for parameters that are references.

File: src/Collections/Parameters.php [hunk 1]
```diff
@@ -50,12 +50,13 @@ protected function doExport()
...
-    public function has(string $name, string $in): bool
+    public function has(string $name, string $in = null): bool
     {
-        return isset($this->parameters[$name.'/'.$in]);
+        $id = $in ? $name.'/'.$in : $name;
+        return isset($this->parameters[$id]);
     }
 
-    public function get(string $name, string $in): Parameter
+    public function get(string $name, string $in = null): Parameter
```
Comment (GuilhemN) [inline src/Collections/Parameters.php]: This looks wrong to me, I would revert. I don't see why we would ever search a parameter by the reference it uses.

File: src/Collections/Parameters.php [hunk 2]
```diff
@@ -50,12 +50,13 @@ protected function doExport()
     /**
      * Searches whether a parameter with the given unique combination exists.
      */
-    public function has(string $name, string $in): bool
+    public function has(string $name, string $in = null): bool
```
Comment (GuilhemN) [inline src/Collections/Parameters.php]: same

File: src/Swagger.php [hunk 3]
```diff
@@ -205,4 +209,12 @@ public function getSecurityDefinitions()
     {
         return $this->securityDefinitions;
     }
+
+    /**
+     * @return array
+     */
+    public function getParameters()
```
Comment (GuilhemN) [inline src/Swagger.php]: No setter?

As a side note, this change requires a new minor. Not a problem for me, I will just release `0.3.0`.

File: src/Collections/Parameters.php (hunk 1 above)
Comment (sandermarechal) [inline src/Collections/Parameters.php]: If I revert, then you could never find that parameter again. E.g:

```php
$operation->getParameters()->add(new Parameter(['$ref' => '#/parameters/foo']));
```
There is no way to check it with `has()` or to `get()` it. Ideally I would look at the target of the reference, then read the `name` and `in` but that is not possible.

File: src/Swagger.php (hunk 3 above)
Comment (sandermarechal) [inline src/Swagger.php]: Sure, I can add a setter.

**Comment by GuilhemN** [comment]
Thank you @sandermarechal!

**Pull request merged**